coffee_core/src/coffee.rs

```diff
@@ -272,11 +273,31 @@ impl PluginManager for CoffeeManager {
         })
     }
 
-    async fn upgrade(&mut self, _: &[&str]) -> Result<(), CoffeeError> {
+    async fn upgrade(&mut self, repo: &str, all: bool) -> Result<CoffeeUpgrade, CoffeeError> {
         self.remote_sync().await?;
-        // FIXME: Fix debug message with the list of plugins to be upgraded
-        log::debug!("upgrading plugins");
-        Ok(())
+        let mut total_status: Vec<CoffeeRepositoryUpgrade> = vec![];
+        if all {
```

Comment:
```suggestion
        if repo.is_none() {
```

and then you can do somethings like that